lib/src/cli.rs

```diff
@@ -348,6 +350,46 @@ enum ProgressOrFinish {
     Finished(Result<ostree_container::Import>),
 }
 
+/// Render an import progress notification as a string.
+pub fn layer_progress_format(p: &ImportProgress) -> String {
+    let (starting, s, layer) = match p {
+        ImportProgress::OstreeChunkStarted(v) => (true, "ostree chunk", v),
+        ImportProgress::OstreeChunkCompleted(v) => (false, "ostree chunk", v),
+        ImportProgress::DerivedLayerStarted(v) => (true, "layer", v),
+        ImportProgress::DerivedLayerCompleted(v) => (false, "layer", v),
+    };
+    // podman outputs 12 characters of digest, let's add 7 for `sha256:`.
+    let short_digest = layer.digest().chars().take(12 + 7).collect::<String>();
+    if starting {
+        let size = glib::format_size(layer.size() as u64);
+        format!("Fetching {s} {short_digest} ({size})")
+    } else {
+        format!("Fetched {s} {short_digest}")
+    }
+}
+
+async fn handle_layer_progress_print(mut r: Receiver<ImportProgress>) {
+    while let Some(v) = r.recv().await {
+        println!("{}", layer_progress_format(&v));
+    }
+}
+
+fn print_layer_status(prep: &PreparedImport) {
+    let (stored, to_fetch, to_fetch_size) =
+        prep.all_layers()
+            .fold((0u32, 0u32, 0u64), |(stored, to_fetch, sz), v| {
```

Comment:
I think this is fine as-is, but it would be easier to read without the same-names shadowing.

Reply:
I didn't initially realize that you named in the same way (`stored` / `to_fetch`) both the outer results and the inner closure parameters, which tripped me for a second. But I don't think this deserves changing, just note that name clashes are not immediate to spot.